src/gpu/nvidia/cudnn_reorder_lt.hpp

```diff
@@ -0,0 +1,232 @@
+
+            // Only support transforming from plain to blocked format and vice-versa.
+            ok = ok && IMPLICATION(src_wrap.is_plain(), !dst_wrap.is_plain());
+            ok = ok && IMPLICATION(dst_wrap.is_plain(), !src_wrap.is_plain());
+            ok = ok && IMPLICATION(!src_wrap.is_plain(), dst_wrap.is_plain());
+            ok = ok && IMPLICATION(!dst_wrap.is_plain(), src_wrap.is_plain());
+            ok = ok && IMPLICATION(src_float_, src_wrap.is_plain());
+            ok = ok && IMPLICATION(dst_float_, dst_wrap.is_plain());
+
+            if (!ok) return ok;
+
+            auto check_tag = [&](const memory_desc_wrapper &wrap,
+                                     bool &transpose, format_kind_t &kind) {
+                kind = format_kind_t::dnnl_blocked;
+                if (wrap.is_cublaslt_blocked_desc()) {
+                    transpose = false;
+                    kind = format_kind::cublaslt_blocked;
+                    return format_tag::undef;
+                }
+                if (wrap.is_plain()) {
+                    auto tag = wrap.matches_one_of_tag(
+                            format_tag::ab, format_tag::abc);
+                    if (tag != format_tag::undef) {
+                        transpose = false;
+                        return tag;
+                    }
+                    tag = wrap.matches_one_of_tag(
+                            format_tag::ba, format_tag::acb);
+                    if (tag != format_tag::undef) {
+                        transpose = true;
+                        return tag;
+                    }
+                }
+                return dnnl_format_tag_undef;
+            };
+            ok = ok && src_wrap.ndims() == dst_wrap.ndims();
+            format_kind_t kind;
+
+            src_tag_ = check_tag(src_wrap, src_trans_, kind);
+            ok = IMPLICATION(
+                    kind == dnnl_blocked, src_tag_ != dnnl_format_tag_undef);
+
+            dst_tag_ = check_tag(dst_wrap, dst_trans_, kind);
+            ok = IMPLICATION(
+                    kind == dnnl_blocked, dst_tag_ != dnnl_format_tag_undef);
+            return ok;
+        }
+
+        bool scales_ok() const {
+            const auto &scales = attr()->scales_;
+            const auto &supported_args = {DNNL_ARG_FROM, DNNL_ARG_TO};
+            if (!scales.has_default_values(supported_args)) return false;
+            // cuDNN does not support scaling per dimension.
+            for (auto arg : supported_args)
+                if (scales.get(arg).mask_ != 0) return false;
+            return true;
+        }
+
+        bool post_ops_ok() const {
+            // only sum post-op is supported
+            const auto &p = attr()->post_ops_;
+            return p.len() == 0 || (p.len() == 1 && p.entry_[0].is_sum(false));
+        }
+
+        status_t init(impl::engine_t *engine, impl::engine_t *src_engine,
+                impl::engine_t *dst_engine) {
```

Comment:
We probably need to add a check that engines are identical

Reply:
Check has been added to ensure both are gpu.